src/overlays/gamestates/ovl_play/m_play.c

```diff
@@ -0,0 +1,868 @@
+    Matrix_MtxtoMtxF(&play->unk_1938.unk_060, &play->unk_1E1C);
+    Skin_Matrix_MulMatrix(&play->unk_1E1C, &play->unk_1E5C, &play->unk_1E1C);
+
+    play->unk_1E5C.mf[0][3] = 0.0f;
+    play->unk_1E5C.mf[1][3] = 0.0f;
+    play->unk_1E5C.mf[2][3] = 0.0f;
+    play->unk_1E5C.mf[3][0] = 0.0f;
+    play->unk_1E5C.mf[3][1] = 0.0f;
+    play->unk_1E5C.mf[3][2] = 0.0f;
+
+    Matrix_reverse(&play->unk_1E5C);
+
+    play->unk_1E9C = _MtxF_to_Mtx(&play->unk_1E5C, (Mtx*)GRAPH_ALLOC(gfxCtx2, sizeof(MtxF) * 1));
+
+    gSPSegment(POLY_OPA_DISP++, 0x01, play->unk_1E9C);
+}
+
+s32 makeBumpTexture(PlayState* play, GraphicsContext* __gfxCtx, GraphicsContext* gfxCtx2) {
+    {
+        Gfx* sp194;
+        Gfx* sp190;
+        sp190 = POLY_OPA_DISP;
+        sp194 = gfxopen(sp190);
+
+        gSPDisplayList(OVERLAY_DISP++, sp194);
+
+        if (play->unk_1EE3 == 3) {
+            PlayState1938 sp60;
+            ScissorViewArg1 sp50;
+
+            initView(&sp60, gfxCtx2);
+            {
+                sp60.unk_120 = 10;
+            label2:;
+            }
+            sp50.unk_04 = 0xF0, sp50.unk_0C = 0x140;
+            sp50.unk_00 = 0;
+            sp50.unk_08 = 0;
+            setScissorView(&sp60, &sp50);
+            showView1(&sp60, 0xF, &sp194);
+            play->unk_1EE8.unk_21C.unk_0C(&play->unk_1EE8, &sp194);
+        }
+
+        fbdemo_fade_draw(&play->unk_2128, &sp194);
+        fade_rgba8888_draw(&sp194, play->unk_220C);
+
+        gSPEndDisplayList(sp194++);
+
+        gfxclose(sp190, sp194);
+        POLY_OPA_DISP = sp194;
+    }
+
+    if (play->unk_1EE2 == 3) {
+        Gfx* sp48 = POLY_OPA_DISP;
+
+        fbdemo_draw(&fbdemo, &sp48);
+        POLY_OPA_DISP = sp48;
+        return 1;
+    }
+
+    PreRender_setup_renderbuf(&play->unk_1DC0, 0x140, 0xF0, gfxCtx2->unk_2E4, gfxCtx2->unk_008);
+    if (play->unk_1CBC.unk_00 == 2) {
+        func_800B0010_jp();
+        PreRender_ConvertFrameBuffer_fg(&play->unk_1DC0);
+        play->unk_1CBC.unk_00 = 3;
+    } else if (play->unk_1CBC.unk_00 >= 5) {
+        play->unk_1CBC.unk_00 = 0;
+    }
+
+    if (play->unk_1CBC.unk_00 == 3) {
+        Gfx* sp44 = POLY_OPA_DISP;
+
+        PreRender_loadFrameBufferCopy(&play->unk_1DC0, &sp44);
+        POLY_OPA_DISP = sp44;
+        return 1;
+    }
+
+    {
+        LightsN* sp40 = Global_light_read(&play->unk_1C60, gfxCtx2);
+
+        LightsN_list_check(sp40, play->unk_1C60.listHead, 0);
+        LightsN_disp(sp40, gfxCtx2);
+    }
+
+    Actor_info_draw_actor(play, &play->actorInfo);
+    Camera2_draw(play);
+    mMsg_Draw(play);
+    Debug_Display_output(play);
+
+    if ((play->unk_1CBC.unk_00 == 1) || (play->unk_1EE2 == 1)) {
+        Gfx* sp3C;
+        Gfx* sp38;
+
+        sp38 = POLY_OPA_DISP;
+        sp3C = gfxopen(sp38);
+
+        gSPDisplayList(OVERLAY_DISP++, sp3C);
+
+        play->unk_1DC0.fbuf = (u16*)gfxCtx2->unk_2E4;
+        play->unk_1DC0.fbufSave = (u16*)gfxCtx2->unk_008;
```

Comment:
Those casts are fishy and don't seem to be required. You should fix the types instead of just silencing the warning

Reply:
cvgSave is a u8* and fbuf/fbufSave are u16*, I don't know which one is right.

Reply:
In cases like this I would recommend to use `UNK_PTR`